tensorbay/opendataset/COCO2017/loader.py

```diff
@@ -0,0 +1,225 @@
+    label.rle = []
+    if image_id not in instances_annotations:
+        return label
+    for annotation in instances_annotations[image_id]:
+        category = categories[annotation["category_id"]]
+        label.box2d.append(LabeledBox2D.from_xywh(*annotation["bbox"], category=category))
+        if annotation["iscrowd"] == 0:
+            points = [list(chunked(coordinates, 2)) for coordinates in annotation["segmentation"]]
+            label.multi_polygon.append(LabeledMultiPolygon(points, category=category))
+        else:
+            label.rle.append(LabeledRLE(annotation["segmentation"]["counts"], category=category))
+    return label
+
+
+def _get_keypoints2d(
+    person_keypoints_annotations: Dict[int, Any], image_id: int, categories: Dict[int, str]
+) -> List[LabeledKeypoints2D]:
+    keypoints2d: List[LabeledKeypoints2D] = []
+
+    if image_id not in person_keypoints_annotations:
+        return keypoints2d
+    for annotation in person_keypoints_annotations[image_id]:
+        points = list(chunked(annotation["keypoints"], 3))
+        category = categories[annotation["category_id"]]
+        keypoints2d.append(LabeledKeypoints2D(points, category=category))
+
+    return keypoints2d
+
+
+def _get_panoptic_box2d(
+    panoptic_annotations: Dict[int, Any], image_id: int, categories: Dict[int, str]
+) -> List[LabeledBox2D]:
+    box2d: List[LabeledBox2D] = []
+
+    if image_id not in panoptic_annotations:
+        return box2d
+    for annotation in panoptic_annotations[image_id]:
+        for segment_info in annotation["segments_info"]:
+            if segment_info["category_id"] > 90:
+                category = categories[segment_info["category_id"]]
+                box2d.append(LabeledBox2D.from_xywh(*segment_info["bbox"], category=category))
+    return box2d
+
+
+def _get_panoptic_mask(
+    annotation_path: str,
+    segment_name: str,
+    image_name: str,
+    panoptic_annotations: Dict[int, Any],
+    image_id: int,
+) -> PanopticMask:
+    compress_pixel, new_mask_path = _save_mask(annotation_path, segment_name, image_name)
+
+    category_ids = {}
+    annotation = panoptic_annotations[image_id][0]
+    for info in annotation["segments_info"]:
+        category_ids[compress_pixel[info["id"]]] = info["category_id"]
+    panoptic_mask = PanopticMask(local_path=new_mask_path)
+    panoptic_mask.all_category_ids = category_ids
+    return panoptic_mask
+
+
+def _save_mask(
+    annotation_path: str, segment_name: str, image_name: str
+) -> Tuple[Dict[int, int], str]:
+    new_file_path = os.path.join(annotation_path, f"new_panoptic_{segment_name}2017")
+    os.makedirs(new_file_path, exist_ok=True)
+
+    array = np.array(
+        Image.open(
+            os.path.join(annotation_path, f"panoptic_{segment_name}2017", f"{image_name}.png")
+        ),
+        dtype=int,
+    )
+    mask_array = array[:, :, 0] + (array[:, :, 1] * 256) + (array[:, :, 2] * 256 * 256)
+
+    array_elements = set(mask_array.flatten())
```

Comment:
set has no order? which will make this function return different results even if the input is the same.

Reply:
how about `np.unique`?